File: testcases/src/main/java/inheritance/two/BaseAbstract.java
package inheritance.two;

import javax.ws.rs.DELETE;
import javax.ws.rs.Path;
import javax.ws.rs.PathParam;

public abstract class BaseAbstract implements ICRUD {

    @Override
    public String createREST(String transferObject) {
        return create(transferObject);
    }



    @Override
    @DELETE
    @Path("delete1/{id1}")
    public String deleteREST( @PathParam("id1") String id) {
        return delete(id);
    }





    private String delete(String id) {
        return id;
    }
    private String create(String transferObject) {
        return transferObject;
    }


}

File: testcases_kt/src/main/kotlin/inheritance/two/BaseAbstract.kt
package inheritance.two

import javax.ws.rs.DELETE
import javax.ws.rs.Path
import javax.ws.rs.PathParam

abstract class BaseAbstract : ICRUD {
    override fun createREST(transferObject: String): String {
        return create(transferObject)
    }

    @DELETE
    @Path("delete1/{id1}")
    override fun deleteREST(@PathParam("id1") id: String): String {
        return delete(id)
    }

    private fun delete(id: String): String {
        return id
    }

    private fun create(transferObject: String): String {
        return transferObject
    }
}
